corehq/motech/repeaters/expression/repeaters.py

```diff
@@ -63,6 +84,35 @@ def get_url(self, repeat_record):
             )
         return base_url
 
+    def handle_response(self, response, repeat_record):
+        super().handle_response(response, repeat_record)
+        if self.update_case_filter_expression and is_response(response):
+            try:
+                self._process_response_as_case_update(response, repeat_record)
+            except Exception as e:
+                notify_exception(None, "Error processing response from Repeater request", e)
+
+    def _process_response_as_case_update(self, response, repeat_record):
+        domain = repeat_record.domain
+        context = get_evaluation_context(domain, repeat_record, self.payload_doc(repeat_record), response)
```

Comment:
nit: It looks like repeat_record is being passed anyway so payload_doc can be fetched in the function itself.

Reply:
That would require `get_evaluation_context` to know the type of the repeat record or have access to the repeater class.